### PR TITLE
feat: Settings js editor

### PR DESCRIPTION
**Pull request by ApekshaBhosale**
## Description

OnPageLoad for js object function

Fixes https://github.com/appsmithorg/appsmith/issues/9130
Fixes https://github.com/appsmithorg/appsmith/issues/9131

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: settings-js-editor 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.68 **(-0.12)** | 37.06 **(-0.14)** | 35.05 **(-0.12)** | 55.99 **(-0.13)**
 :green_circle: | app/client/src/actions/jsPaneActions.ts | 65.52 **(0.3)** | 0 **(0)** | 0 **(0)** | 50 **(0)**
 :red_circle: | app/client/src/actions/pluginActionActions.ts | 67.65 **(-0.03)** | 0 **(0)** | 6.25 **(-0.2)** | 54.55 **(-0.14)**
 :red_circle: | app/client/src/ce/constants/messages.ts | 77.03 **(-0.08)** | 100 **(0)** | 31.85 **(-0.24)** | 82.02 **(-0.26)**
 :red_circle: | app/client/src/components/editorComponents/JSResponseView.tsx | 45.65 **(-8.4)** | 21.98 **(-3.34)** | 0 **(0)** | 49.41 **(-6.15)**
 :red_circle: | app/client/src/entities/DataTree/dataTreeJSAction.ts | 11.76 **(-0.74)** | 0 **(0)** | 0 **(0)** | 9.68 **(-0.66)**
 :sparkles: :new: | **app/client/src/pages/Editor/JSEditor/JSFunctionSettings.tsx** | **52.94** | **100** | **0** | **52.94**
 :red_circle: | app/client/src/reducers/entityReducers/jsActionsReducer.tsx | 7.07 **(-0.72)** | 0 **(0)** | 0 **(0)** | 7.22 **(-0.78)**
 :red_circle: | app/client/src/sagas/EvaluationsSaga.ts | 20.18 **(-0.08)** | 1.94 **(-0.04)** | 11.11 **(0)** | 23 **(-0.12)**
 :red_circle: | app/client/src/sagas/JSPaneSagas.ts | 14.86 **(-2.28)** | 1.11 **(-0.22)** | 4.76 **(-1.49)** | 17.55 **(-2.58)**
 :red_circle: | app/client/src/sagas/PageSagas.tsx | 23.93 **(-0.49)** | 5.38 **(-0.5)** | 13.89 **(-0.82)** | 25.58 **(-0.61)**
 :green_circle: | app/client/src/sagas/ActionExecution/ActionExecutionSagas.ts | 24.75 **(0.57)** | 1.75 **(-0.42)** | 25 **(0)** | 26.04 **(0.75)**
 :red_circle: | app/client/src/sagas/ActionExecution/PluginActionSaga.ts | 18.02 **(-0.36)** | 2.31 **(-0.11)** | 9.52 **(-1.01)** | 20.75 **(-0.37)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :red_circle: | app/client/src/workers/DataTreeEvaluator.ts | 76.33 **(-1.06)** | 57.67 **(-0.78)** | 74.71 **(-0.87)** | 76.05 **(-1.12)**
 :red_circle: | app/client/src/workers/PromisifyAction.ts | 88.37 **(-8.85)** | 90.91 **(0)** | 66.67 **(-33.33)** | 88.57 **(-8.1)**
 :red_circle: | app/client/src/workers/evaluate.ts | 81.58 **(-8.28)** | 37.04 **(-39.88)** | 84 **(0)** | 80.42 **(-8.73)**
 :red_circle: | app/client/src/workers/evaluationUtils.ts | 58.58 **(-0.82)** | 58.79 **(0)** | 65.63 **(0)** | 57.98 **(-0.94)**</details>

### CODE REVIEW

File: app/client/src/sagas/ActionExecution/ActionExecutionSagas.ts [hunk 1]
```diff
@@ -162,6 +178,21 @@ function* initiateActionTriggerExecution(
   // it will be created again while execution
   AppsmithConsole.deleteError(`${source?.id}-${triggerPropertyName}`);
   try {
+    const getEntitySettings = yield call(
+      getActionSettings,
+      action.payload.dynamicString,
+    );
+    const entitySettings = yield call(getConfirmModalFlag, getEntitySettings);
+    if (!!entitySettings) {
+      const confirmed = yield call(confirmRunActionSaga);
```
[the remark this answers is not in the document]
Comment (ApekshaBhosale) [reply app/client/src/sagas/ActionExecution/ActionExecutionSagas.ts:187]: i didn't get it. regex search? @hetunandu

**Review by hetunandu — APPROVED** [review]

**Pull request merged**